dealership/contracts.py

```diff
@@ -1,12 +1,79 @@
+interest_car = 1.07
+interest_motorcycle = 1.03
+interest_truck = 1.11
+employee_discounted_price = 0.9
+lease_multiplier_car = 1.2
+lease_multiplier_motorcycle = 1.0
+lease_multiplier_truck = 1.7
+
+from dealership.vehicles import Car, Truck, Motorcycle
+from dealership.customers import Customer, Employee
+
+
 class Contract(object):
-    pass
+    def __init__(self, vehicle, customer):
+        self.vehicle = vehicle
+        self.customer = customer
 
 
 class BuyContract(Contract):
-    def __init__(self, vehicle, customer, monthly_payments):
-        pass
+    def __init__(self, vehicle, customer, monthly_payments):    
+        super(BuyContract, self).__init__(vehicle, customer)
+        self.monthly_payments = monthly_payments
+  
+    def total_value(self):
+        if isinstance(self.vehicle, Car):
+            car_price = self.vehicle.sale_price() + (interest_car * self.monthly_payments * self.vehicle.sale_price() / 100)
+            if self.customer.is_employee():
+                return car_price * employee_discounted_price
+            else:
+                return car_price
 
+        if isinstance(self.vehicle, Motorcycle):
+            motorcycle_price = self.vehicle.sale_price() + (interest_motorcycle * self.monthly_payments * self.vehicle.sale_price() / 100)
+            if self.customer.is_employee():
+                return motorcycle_price * employee_discounted_price
+            else:
+                return motorcycle_price
+        
+        if isinstance(self.vehicle, Truck):
+            truck_price = self.vehicle.sale_price() + (interest_truck * self.monthly_payments * self.vehicle.sale_price() / 100)
+            if self.customer.is_employee():
+                return truck_price * employee_discounted_price
+            else:
+                return truck_price
 
```

Comment:
noticed there are loads of repetitive codes here which means there's always opportunity to refactor :)

instead of writing it 3 times for `Car`, `Motorcycle` and `Truck`, we can store the price in dictionary like this and reference it later.. 

```
VEHICLE_MULTIPLIERS = {
        Car: 1.07,
        Motorcycle: 1.03,
        Truck: 1.11
    }
```